osu.Game/Graphics/UserInterface/BarGraph.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class BarGraph : FillFlowContainer<Bar>
+    {
+        private BarDirection direction = BarDirection.BottomToTop;
+        public new BarDirection Direction
+        {
+            get
+            {
+                return direction;
+            }
+            set
+            {
+                direction = value;
+                base.Direction = (direction & BarDirection.Horizontal) > 0 ? FillDirection.Vertical : FillDirection.Horizontal;
+                foreach (var bar in Children)
+                {
+                    bar.Size = (direction & BarDirection.Horizontal) > 0 ? new Vector2(1, 1.0f / Children.Count()) : new Vector2(1.0f / Children.Count(), 1);
+                    bar.Direction = direction;
+                }
+            }
+        }
+
+        public IEnumerable<float> Values
+        {
+            set
+            {
+                List<float> values = value?.ToList() ?? new List<float>();
+                List<Bar> graphBars = Children.ToList();
+                for (int i = 0; i < values.Count; i++)

```

Comment:
so when I had 100 values before and now I only give it 99 values  should it remove the extra bar?